### DISCUSSION TITLE
Returning loading state from StreamProvider

### DISCUSSION THREAD

**Discussion by kentnek** [rrousselGit/riverpod#1530] [Q&A] [answered]
Hi, I'm using `StreamProvider` to provide data retrieved from Firebase Realtime DB (RTDB):

```dart
final dataFromFirebaseProvider = StreamProvider.autoDispose<SomeModel>((ref) {
  final database = ref.read(realtimeDatabaseProvider);

  return database.ref("some_key").onValue
    .map((event) {
      if (!event.snapshot.exists) {
        // return AsyncValue.loading()?
      }

      final data = event.snapshot.value as Map<dynamic, dynamic>;
      return SomeModel(...); // convert data to SomeModel
    });
});
```

I notice that updating `some_key` in RTDB will trigger two events, one with `snapshot.exists == false` (`some_key` being deleted in RTDB) before another event with the actual updated data.

How do I return `AsyncValue.loading()` when the first event is encountered? Thanks!

**Comment by kentnek** [comment]
@rrousselGit may I ask for an update? Thanks!

**Comment by rrousselGit** [comment, marked as answer]
in 1.0 you can't. In 2.0 you could do `ref.state = AsyncLoading()`

**Reply by kentnek** [reply]
awesome, thank you so much!
